src/bindgen/cargo/cargo_expand.rs

```diff
@@ -82,6 +82,10 @@ pub fn expand(
        cmd.env("CARGO_TARGET_DIR", out_dir.join("expanded"));
     }
 
+    // Set this variable so that we don't call it recursively if we expand a crate that is using
+    // cbindgen
+    cmd.env("_CBINDGEN_IS_RUNNING", "1");
```

Comment:
This is a bit hacky... Though I don't have better ideas.

Reply:
Yes, it's super hacky, but it's also the best idea I had.